### PR TITLE
Restore support for version in dataset version

### PR DESCRIPTION
**Pull request by anandjakhaniya**
**Jira Ticket:** https://vertaai.atlassian.net/browse/VR-6256

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java [hunk 1]
```diff
@@ -211,6 +226,37 @@ public CommitDAORdbImpl(AuthService authService, RoleService roleService) {
               .setIdType(IDTypeEnum.IDType.VERSIONING_REPO_COMMIT_BLOB)
               .build(),
           datasetVersion.getTagsList());
+
+      long version = datasetVersion.getVersion();
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java:230]: if the user sets a version we should use it.

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java [hunk 2]
```diff
@@ -211,6 +226,37 @@ public CommitDAORdbImpl(AuthService authService, RoleService roleService) {
...
+                    .setIdType(IDTypeEnum.IDType.VERSIONING_REPO_COMMIT_BLOB)
+                    .setStringId(parentCompositeId)
+                    .build(),
+                ModelDBConstants.VERSION);
+        if (parentVersion != null && !parentVersion.isEmpty()) {
+          version = Long.parseLong(parentVersion) + 1L;
+        }
+      }
+      if (version == 0) {
+        version = version + 1;
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]: so version = 1 ??? 

we should only look at the version of parent commit or even fetch parent commit if there is no version set explicitly

```
if the request has a version save it , else increment the version of the parent commit and store it
```

**Review by ravishetye — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]: Change it according to comment 
```
if the request has a version save it, else increment the version of the parent commit and store it
```

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java (hunk 1 above)
Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java:230]: I can not set a version from the request, was that flow verified.

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java:230]: fixed

**Pull request merged**